File: 08-2-2-Adapter-Solution/Ducks/SuperDuck.cs
namespace Adapter.Ducks
{
    using System;
    using Adapter.Fly;
    using Adapter.Quack;

    public sealed class SuperDuck : Duck
    {
        private readonly string _appearance;

        public SuperDuck(IQuackStrategy quackStrategy, IFlyStrategy flyStrategy, string appearance)
            : base(quackStrategy, flyStrategy)
        {
            _appearance = appearance;
        }

        public override void Display() => Console.WriteLine(_appearance);
    }
}

File: 08-DesignPatterns-Java/Solution/Strategy/src/Ducks/SuperDuck.java
package Ducks;

import Fly.IFlyStrategy;
import Quack.IQuackStrategy;

public class SuperDuck extends Duck {
    private final String _appearance;

    public SuperDuck(IQuackStrategy quackStrategy, IFlyStrategy flyStrategy, String appearance) {
        super(quackStrategy, flyStrategy);
        _appearance = appearance;
    }

    @Override
    public void display() {
        System.out.println(_appearance);
    }
}
